mod-user-import/src/main/resources/domain/mod-user-import/features/userImport.feature

```diff
@@ -19,36 +25,69 @@ Feature: User import
     When method POST
     Then status 200
 
+  # Importing a user requires that the group associated with the user already be created.
+  Scenario: Add user group for tenant
+    Given path 'groups'
+    And request
+    """
+    {
+      "group": "#(userGroup)",
+      "desc": "A user group.",
+    }
+    """
+    When method POST
+    Then status 201
+    Then match response.group == userGroup
+
+  Scenario: Get groups for tenant and verify
+    Given path 'groups'
+    When method GET
+    Then status 200
+    And assert response.usergroups.length == 1
+    And assert response.totalRecords == 1
+
   # Import a set of users as defined in a JSON array by posting to the endpoint.
-  @Undefined
-  Scenario: Import with JSON users array and check JSON response
-    * print 'undefined'
-    # TODO Check the user import response. Do the following properties have the correct value in the JSON response:
-    # * createdRecords
-    # * updatedRecords
-    # * failedRecords
-    # * totalRecords
-    # * ?
-
-  # Fetch a given user that was imported from JSON array and see if properties match the user's properties in the array.
-  @Undefined
-  Scenario: Verify JSON user import for a given user
-    * print 'undefined'
-
-  # Update a set set of users as defined in a JSON array by posting to the endpoint.
-  @Undefined
-  Scenario: Update with JSON users array and check JSON response
-    * print 'undefined'
-    # TODO Check the user update response. Do the following properties have the correct value in the JSON response:
-    # * createdRecords
-    # * updatedRecords
-    # * failedRecords
-    # * totalRecords
-    # * ?
-
-  # Fetch a given user that was updated from the JSON array and see if properties match the original user's properties.
-  @Undefined
-  Scenario: Verify JSON user import for a given user
-    * print 'undefined'
+  Scenario: Import new users with JSON users array and check JSON response
+    Given path 'user-import'
+    And request usersToImport
+    When method POST
+    Then status 200
+    And match response.createdRecords == 1
+    And match response.updatedRecords == 0
+    And match response.failedRecords == 0
+    And match response.totalRecords == 1
 
+  Scenario: Verify user was imported successfully by getting the user and checking its properties
+    Given path 'users'
+    And param query = "barcode==" + barcode
 
+    When method GET
+    Then status 200
+    And assert response.users[0].barcode == barcode
+    And assert response.users[0].username == usersToImport.users[0].username
+    And assert response.totalRecords == 1
+    # Assign the user id for use in subsequent scenarios. The id is created when the user is inserted.
+    * def userId = response.users[0].id
```

Comment:
Apparently sharing variables between scenarios (what I'm doing here) [is not](https://github.com/intuit/karate#script-structure) the "karate way". Want to refactor this so that each scenario stands on its own.

Reply:
This is no longer an issue since the update is happening using the externalSystemId.